rfc#0001

```diff
@@ -0,0 +1,145 @@
+# WASM binary loading into an Enarx Keep
+- Authors: [Mike Bursell](https://github.com/MikeCamel)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-01-10
+- Status Note: (has been subject to initial discussion and consensus)  
+- Supersedes: n/a
+- Start Date: 2020-02-16
+- Tags: architecture, webassembly
+
+## Summary
+
+This RFC describes the architecture adopted by Enarx for loading Wasm
+packages into a prepared Keep using a secure channel. 
+
+## Motivation
+
+Once a Keep is prepared with microkernel, Wasm runtime, etc., the
+tenant workload (Wasm binary + data) needs to be loaded into it.  A
+description of the loading sequence is required.
+
+## Tutorial
+
+A Keep is the fundamental runtime component of Enarx: a TEE instance
++ Enarx runtime pieces, including WebAssembly and WASI layers.  In
+order for the Keep to execute a workload, the Wasm (WebAssembly) package
+that comprises the workload (binary/binaries + data elements) need to be loaded
+into the Keep.  This workload is sent to the Keep by the Enarx client agent
+(ECA), encrypted under a session key (the workload provisioning encryption
+key or WPEK) that is established at Keep provisioning time and
+communicated to the ECA.
```

Comment:
```suggestion
communicated between the ECA and the Keep using a Diffie-Hellman Key Exchange or similar.
```
I still think it would be better to word this more explicitly. "communicated to" is a bit vague as there is no subject and someone might infer that some third entity aside from the ECA and Keep is involved, or that the Keep is solely in charge of generating the key (which also isn't the case, at least the way we used it with SGX in the early demo).

Reply:
This is dealt with in detail in a separate RFC.  I've improved the wording somewhat.